ksqldb-parser/src/main/java/io/confluent/ksql/util/QueryMask.java

```diff
@@ -0,0 +1,166 @@
+        matcher.appendReplacement(sb, replacement);
+      } else {
+        matcher.appendReplacement(sb, matchedString);
+      }
+    }
+
+    matcher.appendTail(sb);
+    return sb.toString();
+  }
+
+  private static final class Visitor extends SqlBaseBaseVisitor<String> {
+
+    private boolean masked;
+
+    @Override
+    public String visitStatements(final StatementsContext context) {
+      final List<String> statementList = new ArrayList<>();
+      for (final SingleStatementContext stmtContext : context.singleStatement()) {
+
+        masked = false;
+        final String statement = visitSingleStatement(stmtContext);
+
+        if (masked) {
+          statementList.add(statement);
+        } else {
+          // Use original statement if not masked
+          final int start = stmtContext.start.getStartIndex();
+          final int end = stmtContext.stop.getStopIndex();
+          final Interval interval = new Interval(start, end);
+          final String original = stmtContext.start.getInputStream().getText(interval);
+          statementList.add(original);
+        }
+      }
+      return StringUtils.join(statementList, "\n");
+    }
+
+    @Override
+    public String visitSingleStatement(final SingleStatementContext context) {
+      return String.format("%s;", visit(context.statement()));
+    }
+
+    @Override
+    public String visitCreateConnector(final CreateConnectorContext context) {
+      final StringBuilder stringBuilder = new StringBuilder("CREATE");
+
+      if (context.SOURCE() != null) {
+        stringBuilder.append(" SOURCE");
+      } else if (context.SINK() != null) {
+        stringBuilder.append(" SINK");
+      }
+
+      stringBuilder.append(" CONNECTOR");
+
+      // optional if not exists
+      if (context.EXISTS() != null) {
+        stringBuilder.append(" IF NOT EXISTS");
+      }
+
+      stringBuilder.append(" ").append(context.identifier().getText());
+
+
+      // mask properties
+      if (context.tableProperties() != null) {
+        stringBuilder.append(visit(context.tableProperties()));
+      }
+
+      return stringBuilder.toString();
+    }
+
+    @Override
+    public String visitTableProperties(final TablePropertiesContext context) {
+      if (!(context.parent instanceof CreateConnectorContext)) {
+        return null;
+      }
+      final List<String> tableProperties = new ArrayList<>();
+      // Mask everything except "connector.class"
+      for (final TablePropertyContext prop : context.tableProperty()) {
+        final StringBuilder formattedProp = new StringBuilder();
+        final String unquotedLowerKey;
+        final String key;
+
+        if (prop.identifier() != null) {
+          key = prop.identifier().getText();
+          unquotedLowerKey = ParserUtil.getIdentifierText(prop.identifier()).toLowerCase();
+        } else {
+          key = prop.STRING().getText();
+          unquotedLowerKey = key.substring(1, key.length() - 1).toLowerCase();
```

Comment:
nit: use `ParserUtil.unquote(key, "'")` instead to match what AstBuilder does.

Reply:
I'm using `key.substring(1, key.length() - 1).toLowerCase();` since it could double quotes and single quotes?

Reply:
If it's double quotes or backticks it should be caught under the "identifier" case above, rather than the "string literal" case that this is handling.